### PR TITLE
[REFACTOR] refactor login middleware

### PR DESCRIPTION
**Pull request by jasonirvine76**
This PR introduces significant improvements to the Auth Middleware in the project, addressing maintainability and overall code quality.

Key Changes:
**Middleware Refactor**
- Improved code quality

**Test**
- Adapt with the new auth middleware

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -151,6 +151,15 @@
     slack_app.event("reaction_added")(slack_adapter.reaction_added)
 
     app = FastAPI()
+    app.add_middleware(AuthMiddleware, jwt_secret_key=config.jwt_secret_key, included_routes=[
```
Comment (Veivel) [inline main.py:154]: Could you clarify what would need to be done to use the login middleware for other endpoints? Do you just add the path here?

Comment (jasonirvine76) [reply main.py:154]: Yes, just add the path there

Comment (Veivel) [reply main.py:154]: OK thanks for the info!

**Review by Veivel — APPROVED** [review]
LGTM! just praying for no merge conflicts on my end 🙏

**Pull request merged**